### DISCUSSION TITLE
Inconsistent Typescript behavior

### DISCUSSION THREAD

**Discussion by mauzybwy** [TanStack/query#6795] [Q&A] [answered]
I have two projects, both using @tanstack/react-query@15.18.0 with typescript@5.3.3, however Typescript is inferring types differently.

Take the following example:
```
const { data } = useQuery({
  queryKey: ["test"],
  queryFn: () => Promise.resolve(5),
});
```

Project A correctly reports `const data: number | undefined`, while project B reports `const data: number`. Does anybody have insight into this issue?

**Comment by TkDodo** [comment, marked as answer]
different tsconfig settings for `strict` mode ?

**Reply by mauzybwy** [reply]
Yes, forgot to update question when I figured that out. But that was indeed the issue.
